### PR TITLE
Zoom out to show more results and Zoom to city when filter by city

### PR DESCRIPTION
**Pull request by goemen**

### CODE REVIEW

File: backend/dal/Helpers/Extensions/PropertyExtensions.cs [hunk 1]
```diff
@@ -99,6 +99,9 @@ public static IQueryable<Entity.Views.Property> GenerateQuery(this PimsContext c
             if (!String.IsNullOrWhiteSpace(filter.Address))
                 query = query.Where(p => EF.Functions.Like(p.Address, $"%{filter.Address}%") || EF.Functions.Like(p.City, $"%{filter.Address}%"));
 
+            if (!String.IsNullOrWhiteSpace(filter.City))
+                query = query.Where(p => EF.Functions.Like(p.City, $"%{filter.City}%") || EF.Functions.Like(p.City, $"%{filter.City}%"));
```
Comment (Fosol) [inline backend/dal/Helpers/Extensions/PropertyExtensions.cs:103]: You've duplicated the LIKE statement.  Remove the OR.

File: backend/entities/Models/BuildingFilter.cs [hunk 2]
```diff
@@ -81,9 +82,10 @@ public BuildingFilter(double neLat, double neLong, double swLat, double swLong)
         /// <param name="maxAssessedValue"></param>
         /// <param name="sort"></param>
         /// <returns></returns>
-        public BuildingFilter(string address, int? agencyId, int? constructionTypeId, int? predominantUseId, int? floorCount, string tenancy, float? minRentableArea, float? maxRentableArea, decimal? minEstimatedValue, decimal? maxEstimatedValue, decimal? minAssessedValue, decimal? maxAssessedValue, string[] sort)
+        public BuildingFilter(string address, string city, int? agencyId, int? constructionTypeId, int? predominantUseId, int? floorCount, string tenancy, float? minRentableArea, float? maxRentableArea, decimal? minEstimatedValue, decimal? maxEstimatedValue, decimal? minAssessedValue, decimal? maxAssessedValue, string[] sort)
```
Comment (Fosol) [inline backend/entities/Models/BuildingFilter.cs:85]: This isn't really necessary I suspect.

File: backend/entities/Models/PropertyFilter.cs [hunk 3]
```diff
@@ -145,9 +152,10 @@ public PropertyFilter(double neLat, double neLong, double swLat, double swLong)
         /// <param name="maxAssessedValue"></param>
         /// <param name="sort"></param>
         /// <returns></returns>
-        public PropertyFilter(string address, int? agencyId, int? classificationId, decimal? minEstimatedValue, decimal? maxEstimatedValue, decimal? minAssessedValue, decimal? maxAssessedValue, string[] sort)
+        public PropertyFilter(string address, string city, int? agencyId, int? classificationId, decimal? minEstimatedValue, decimal? maxEstimatedValue, decimal? minAssessedValue, decimal? maxAssessedValue, string[] sort)
```
Comment (Fosol) [inline backend/entities/Models/PropertyFilter.cs:155]: This isn't necessary I suspect.

**Comment by devinleighsmith** [comment]
Does this PR address zooming out when none of the filtered results are displayed in the current viewpoint? or is that a different issue?

**Comment by goemen** [comment]
> Does this PR address zooming out when none of the filtered results are displayed in the current viewpoint? or is that a different issue?
@devinleighsmith it is part of this ticket

**Comment by devinleighsmith** [comment]
@goemen could you point me to the part in this PR that handles jumping to a search result if none are being displayed in the current bounds? reading through it is wasn't obvious to me what code was handling that.

**Review by Fosol — APPROVED** [review]

**Comment by goemen** [comment]
> @goemen could you point me to the part in this PR that handles jumping to a search result if none are being displayed in the current bounds? reading through it is wasn't obvious to me what code was handling that.

File: frontend/src/components/maps/leaflet/Map.tsx [hunk 4]
```diff
@@ -89,6 +92,17 @@ const Map: React.FC<MapProps> = ({
   });
   const [baseLayers, setBaseLayers] = useState<BaseLayer[]>([]);
   const [activeBasemap, setActiveBasemap] = useState<BaseLayer | null>(null);
+  const smallScreen = useMediaQuery({ maxWidth: 1800 });
+
+  useEffect(() => {
```
Comment (goemen) [inline frontend/src/components/maps/leaflet/Map.tsx:97]: @devinleighsmith this is the code that will trigger the zoom out to fetch more results

**Pull request closed without merging**